dev/set_matrix.py

```diff
@@ -133,6 +133,35 @@ def is_final_release(ver):
     return re.search(r"^\d+(\.\d+)+$", ver) is not None
 
 
+def is_post_release(ver):
```

Comment:
I found [packaging](https://packaging.pypa.io/en/latest/version.html#usage) provides this functionality.

Reply:
Would be awesome if we could drop in `packaging` instead, but it's not a deal breaker if that's too difficult.

Reply:
I'll merge this PR after a small refactor.